video/out/vo_gpu_next.c

```diff
@@ -991,27 +991,34 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
         p->last_id = id;
     }
 
+    struct ra_swapchain *sw = p->ra_ctx->swapchain;
+
     bool pass_colorspace = false;
+    struct pl_color_space target_csp;
+    target_csp = sw->fns->target_csp ? sw->fns->target_csp(sw) : (struct pl_color_space){0};
+    float target_peak = opts->target_peak ? opts->target_peak : target_csp.hdr.max_luma;
+
     struct pl_color_space hint;
-    if (p->next_opts->target_hint && frame->current) {
+    bool target_hint = p->next_opts->target_hint == 1 ||
+                       (p->next_opts->target_hint == -1 && target_csp.transfer == PL_COLOR_TRC_PQ);
```

Comment:
Shouldn't this be `pl_color_transfer_is_hdr(target_csp.transfer)`?

Reply:
Sure, changed.